src/app/services/card.service.ts

```diff
@@ -41,29 +41,61 @@ export interface Card {
     color: string;
     owner: string;
     preview?: string;
+    preview$?: Observable<string>;
+    previewId?: string;
+    organisation?: string;
+    isPublic?: boolean;
 }
 
 @Injectable({
     providedIn: 'root'
 })
 export class CardService {
 
-    public constructor(private persistenceService: PersistenceService,
+    public constructor(
+        private persistenceService: PersistenceService,
         private colorService: ArlasColorService) {
     }
 
-    public cardList(size: number, page: number, org: string): Observable<[number, Card[]]> {
-        return (this.persistenceService.list('config.json', size, page, 'desc') as Observable<DataResource>)
-            .pipe(map(data => {
-                if (data.data !== undefined) {
-                    return [data.total, data.data.map(d => this.dataWithlinksToCards(d, org))];
+    public cardList(options?: any): Observable<Card[]> {
+        // First call to list to find the total number of dashboards
+        return this.getListObs(1, options)
+            .pipe(mergeMap((one) => {
+                if (one.data !== undefined) {
+                    // Second call to list to retrieve all the  dashboards
+                    return this.getListObs(one.total, options);
                 } else {
-                    return [data.total, []];
+                    return of({});
+                }
+            }), map(data => {
+                if ((data as any).data !== undefined) {
+                    return (data as DataResource).data.map(d => this.dataWithlinksToCard(d));
+                } else {
+                    return [];
                 }
             }));
     }
 
-    private dataWithlinksToCards(data: DataWithLinks, org: string): Card {
+    private getListObs(size, options?): Observable<DataResource> {
+        return this.persistenceService.list('config.json', size, 1, 'desc', options);
+    }
+
+    private getMapCard(data: DataWithLinks[]): Map<string, Card[]> {
+        const dataMap = new Map<string, Card[]>();
+        data.map(d => this.dataWithlinksToCard(d)).forEach(c => {
+            const cardsForOrga = dataMap.get(c.organisation);
+            if (!!cardsForOrga) {
+                cardsForOrga.push(c);
+            } else {
+                dataMap.set(c.organisation, []);
+                dataMap.get(c.organisation).push(c);
+            }
+        });
+        return dataMap;
+    }
+
+    private dataWithlinksToCard(data: DataWithLinks): Card {
+        const organisation = (!!data.doc_organization || data.doc_organization !=='')  ? data.doc_organization : 'no_organisation';
```

Comment:
'no_organisation' is widely used across all projects built on the toolkit, it should be a constant there